shell/ci/release/confluence-publish.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+# This script publishes all markdown files with certain directives to confluence
+# using said directives.
+#
+# Directives:
+# <!-- Space: <confluence space key> --> (required)
+# <!-- Parent: <title of parent page> --> (not required)
+# <!-- Title: <title of page to publish to> --> (required)
+
+set -e
+
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+LIB_DIR="${DIR}/../../lib"
+
+# shellcheck source=../../lib/logging.sh
+source "${LIB_DIR}/logging.sh"
+
+# shellcheck source=./lib/bootstrap.sh
+source "$DIR/lib/bootstrap.sh"
+
+srcPath=$DIR
+info "srcPath: ${DIR}"
+
+tag=$(eval "git describe --tags --abbrev=0")
+findCmd="git diff-tree --no-commit-id --name-status -r ${tag} HEAD | grep -e '^[A|M].*\.md$' | cut -f2"
+
+info "findCmd: ${findCmd}"
+
+defaultBranch="$(git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@')"
+
+info "defaultBranch: ${defaultBranch}"
+
+retry() {
```

Comment:
We should already have a retry helper in `shell/lib/shell.sh` (something along those lines?). If we don't, move that into it's own lib.